pkg/connector/tickets.go

```diff
@@ -0,0 +1,464 @@
+		DisplayName:  project.Name,
+		Types:        ticketTypes,
+		CustomFields: customFields,
+	}
+
+	jiraStatuses, err := getJiraStatusesForProject(ctx, j.client, project.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	// iterate through statues, if global or done or projectId
+	statuses, err := j.getTicketStatuses(ctx, project.ID, jiraStatuses)
+	if err != nil {
+		return nil, err
+	}
+	ret.Statuses = statuses
+
+	j.ticketSchemas[project.Key] = ret
+
+	return ret, nil
+}
+
+func (j *Jira) GetTicketSchema(ctx context.Context, schemaID string) (*v2.TicketSchema, annotations.Annotations, error) {
+	if schema, ok := j.ticketSchemas[schemaID]; ok {
+		return schema, nil, nil
+	}
+
+	project, _, err := j.client.Project.Get(ctx, schemaID)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ret, err := j.schemaForProject(ctx, *project)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return ret, nil, nil
+}
+
+func (j *Jira) issueToTicket(ctx context.Context, issue *jira.Issue) (*v2.Ticket, error) {
+	if issue.Fields == nil {
+		return nil, errors.New("issue has no fields")
+	}
+	schema, _, err := j.GetTicketSchema(ctx, issue.Fields.Project.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	issueURL, err := j.generateIssueURL(issue.Key)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &v2.Ticket{
+		Id:          issue.ID,
+		DisplayName: issue.Fields.Summary,
+		Description: issue.Fields.Description,
+		Type: &v2.TicketType{
+			Id:          issue.Fields.Type.ID,
+			DisplayName: issue.Fields.Type.Name,
+		},
+		Status: &v2.TicketStatus{
+			Id:          issue.Fields.Status.ID,
+			DisplayName: issue.Fields.Status.Name,
+		},
+		Labels:    issue.Fields.Labels,
+		CreatedAt: timestamppb.New(time.Time(issue.Fields.Created)),
+		UpdatedAt: timestamppb.New(time.Time(issue.Fields.Updated)),
+		Url:       issueURL,
+	}
+
+	if issue.Fields.Assignee != nil {
+		if assignee, err := userResource(ctx, issue.Fields.Assignee); err == nil && assignee != nil {
+			ret.Assignees = []*v2.Resource{assignee}
+		}
+	}
+
+	if issue.Fields.Reporter != nil {
+		if reporter, err := userResource(ctx, issue.Fields.Reporter); err == nil && reporter != nil {
+			ret.Reporter = reporter
+		}
+	}
+
+	retCustomFields := make(map[string]*v2.TicketCustomField)
+	for id, cf := range schema.GetCustomFields() {
+		switch id {
+		case "project":
+			retCustomFields[id] = sdkTicket.PickObjectValueField(cf.GetId(), &v2.TicketCustomFieldObjectValue{
+				Id:          issue.Fields.Project.ID,
+				DisplayName: issue.Fields.Project.Name,
+			})
+		case "components":
+			var components []*v2.TicketCustomFieldObjectValue
+			for _, component := range issue.Fields.Components {
+				components = append(components, &v2.TicketCustomFieldObjectValue{
+					Id:          component.ID,
+					DisplayName: component.Name,
+				})
+			}
+			retCustomFields[id] = sdkTicket.PickMultipleObjectValuesField(cf.GetId(), components)
+		}
+	}
+	ret.CustomFields = retCustomFields
+
+	return ret, nil
```

Comment:
this previously was validating the ticket but we we're doing it twice in create ticket